testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -7126,6 +7126,9 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
         "Cannot apply 'ARRAYS_OVERLAP' to arguments of type 'ARRAYS_OVERLAP\\("
             + "<INTEGER ARRAY>, <BOOLEAN>\\)'. Supported form\\(s\\): 'ARRAYS_OVERLAP\\("
             + "<EQUIVALENT_TYPE>, <EQUIVALENT_TYPE>\\)'", false);
+    f.checkNull("arrays_overlap(null, null)");
```

Comment:
@mihaibudiu Are you sure spark supports this writing method? My test failed in spark 3.4.

Reply:
![2024-04-02 23-10-05屏幕截图](https://github.com/apache/calcite/assets/77189278/0c9fd0bb-2953-431b-a4e7-87d57618b96b)

Reply:
I am not sure what you mean by that. This is a legal expression which shouldn't crash the compiler.

Reply:
![2024-04-02 23-10-05屏幕截图](https://github.com/apache/calcite/assets/77189278/bf7bd82a-f3a7-4a46-9a90-a24e956b8892)

Reply:
![2024-04-03 08-20-47屏幕截图](https://github.com/apache/calcite/assets/77189278/087df7b0-e465-498b-8c9e-0f49ec29db8c)

Reply:
> I am not sure what you mean by that. This is a legal expression which shouldn't crash the compiler.

It seems to be a bug in spark. I can help you test it on the latest code branch of spark to see if it can be fixed.

Reply:
@mihaibudiu I tested it on the spark master branch. It seems that the parameters in arrays_overlap are not allowed to be null, which will cause the program to crash.

Reply:
which program crashes?

Reply:
There may be something wrong with my expression. It should not be said that the program crashed, but that the execution failed. sorry
![2024-04-03 09-04-51屏幕截图](https://github.com/apache/calcite/assets/77189278/8cbdca79-dc2d-4e45-bd74-c34881034437)

Reply:
Whoever wrote that code is confusing the type "NULL" with the type "VOID". These are not the same thing at all. NULL is a type which has one value, whereas VOID has no values at all.
So there is at least one bug in their handling of types.
We can also reject this expression at compilation time in Calcite.
What happens if both arrays are NULL?

Reply:
The situation of arrays_overlap(null, null) is as follows
![2024-04-03 09-12-28屏幕截图](https://github.com/apache/calcite/assets/77189278/93d71cb4-ee29-446f-8508-bb01e1564efa)

Reply:
then it's easy, we just reject arguments of type NULL for this function.
I will rework the implementation.